plugins/out_s3/s3.c

```diff
@@ -83,6 +83,27 @@ int s3_plugin_under_test()
     return FLB_FALSE;
 }
 
+struct flb_aws_header *create_canned_acl_header(char *canned_acl)
+{
+    struct flb_aws_header *h = NULL;
+    
+    if (canned_acl == NULL) {
+        return NULL;
+    }
```

Comment:
You can keep this, but presumably the caller of this function would check if the ACL is null or not- i.e. once this function is called you can assume its not null

Reply:
I didn't check if it is NULL when I call it. Do I need to implement the code in that way? I want the logic as `if (canned_acl == NULL || flb_malloc doesn't work)`, we return NULL.

Reply:
Yeah, see my other comment on separation of responsibility, and giving each function a "job" which is as small and clear as possible